app/controllers/Management.scala

```diff
@@ -40,10 +43,23 @@ object Management extends Controller with LazyLogging {
     override def ok = exec()
   }
 
+  class EitherTest[A](name: String, exec: () => List[String] \/ A) extends Test {
+    override def messages = exec().swap.toOption.toList.flatten
+    override def ok: Boolean = exec().isRight
```

Comment:
or return the whole thing as an object and use this as a factory, if it does need reevaluating every time it's used as a whole